lib/blocks.php

```diff
@@ -45,3 +45,58 @@ function gutenberg_reregister_core_block_types() {
 	}
 }
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
+
+/**
+ * Registers a new block style.
+ *
+ * @param string $block_name       Block type name including namespace.
+ * @param array  $style_properties Array containing the properties of the style name, label, style (name of the stylesheet to be enqueued), inline_style (string containing the CSS to be added).
+ */
+function gutenberg_register_block_style( $block_name, $style_properties ) {
+	if ( isset( $style_properties[ 'style' ] ) ) {
+		$style_handle = $style_properties[ 'style' ];
+		unset( $style_properties[ 'style' ] );
+		add_action(
+			'enqueue_block_assets',
+			function() use ( $style_handle ) {
+				wp_enqueue_style( $style_handle );
+			},
+			30
+		);
+	}
+
+	if ( isset( $style_properties[ 'inline_style' ] ) ) {
+		$inline_style = $style_properties[ 'inline_style' ];
+		unset( $style_properties[ 'inline_style' ] );
+		add_action(
+			'enqueue_block_assets',
+			function() use ( $inline_style ) {
+				wp_add_inline_style( 'wp-block-library', $inline_style );
```

Comment:
What if it's not a core block?

Reply:
What problem would that cause?

Reply:
I mean yeah attaching to `wp-block-library` seems weird. Maybe `wp-block-editor` instead but I don't see any problem resulting from this.

The question is what does it mean to register a block style. Are we registering this style for all block-editor implementations in core, just the post editor..

Reply:
(also same question about the script down here)

Reply:
I think using wp-block-editor would not work because these styles also need to be enqueued on the website frontend.
Ideally, these styles would be loaded after the block styles because, in case of specificity conflict, block styles would be the used ones. Using wp-block-library ensures that this styles load after the core block styles, it is true that for non-core ones that's not the case, I guess we can increase that probability by changing to a lower priority.
For non-core blocks, an option would be to use WP_Block_Type_Registry get the registered block and add use `wp_add_inline_style( $block_type->style, $inline_style );`. The problem is that I verified and for most cases this style is null. I guess it is only not null if the block was registered with register_block_type PHP function and the style was passed as one of the args -- I think that this not a common way to register blocks and its styles. 

Another option would be to register an inline styles handle that does not point to a file:
wp_register_style( 'wp-inline-block-styles', false );
wp_enqueue_style( 'wp-inline-block-styles' );

And then we would use wp_add_inline_style( 'wp-inline-block-styles', $inline_style );

The current option ensures that inline-block styles are loaded when wp-block-library styles are loaded if these styles are enqueued inline-block styles would also not be added, that seemed a good option.
Using a handle just for the inline-block styles also does not seems wrong, but using false for src looks a little "hacky".


Regarding the inline script to register the style, I used wp-blocks as its handle because it calls a function part of wp.blocks.

Reply:
Thanks for the details, it seems like there's no perfect solution at the moment. Let's ship the first version as is and iterate.